components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/new_tab_page_ads/eligible_new_tab_page_ads_v2.cc

```diff
@@ -133,10 +134,15 @@ CreativeNewTabPageAdList EligibleNewTabPageAdsV2::FilterCreativeAds(
     return {};
   }
 
+  CreativeNewTabPageAdList eligible_creative_ads = creative_ads;
```

Comment:
This is an extra copy. Maybe it is better to do:

```
  CreativeInlineContentAdList eligible_creative_ads = ApplyExclusionRules(
      creative_ads, last_served_ad_, &exclusion_rules);

```

Reply:
Great catch. Fixed.